ci/code_checks.sh

```diff
@@ -48,31 +48,6 @@ fi
 ### LINTING ###
 if [[ -z "$CHECK" || "$CHECK" == "lint" ]]; then
 
-    echo "black --version"
```

Comment:
how are these no longer needed?

Reply:
They're run during the pre-commit CI check, see https://github.com/pandas-dev/pandas/pull/36471/checks?check_run_id=1137265216

Reply:
but do we run the pre-commit checks on CI? i think have a script that is separate from the pre-commit is needed as well (it can duplicate or be called by pre-commit), but key is to have everything in sync.

Reply:
Two cents: I think keeping this for a time just to ensure it never disagrees with pre-commit makes sense, then can remove later once we're fairly sure they're identical (also makes it easier to revert if misconfigured)

Reply:
> but do we run the pre-commit checks on CI?

That's what this PR adds - you go down to "All checks have passed" and expand, you'll see an extra GitHub Action called "pre-commit". If you expand that one further, it says

```
Run pre-commit/action@v2.0.0
install pre-commit
Cache Size: ~21 MB (21895398 B)
/bin/tar --use-compress-program zstd -d -xf /home/runner/work/_temp/f09139a0-b8b5-427f-bc76-10b2403b8816/cache.tzst -P -C /home/runner/work/pandas/pandas
/opt/hostedtoolcache/Python/3.8.5/x64/bin/pre-commit run --show-diff-on-failure --color=always --all-files
black....................................................................Passed
flake8...................................................................Passed
flake8-pyx...............................................................Passed
flake8-pxd...............................................................Passed
isort....................................................................Passed
pyupgrade................................................................Passed
```

Reply:
> Two cents: I think keeping this for a time just to ensure it never disagrees with pre-commit makes sense, then can remove later once we're fairly sure they're identical (also makes it easier to revert if misconfigured)


This seems like a good approach! Also would be good to check that github actions CI fails if any of these pre commit stages doesn’t pass.

Reply:
Sure - have added the ci/code checks back in for now then

> Also would be good to check that github actions CI fails if any of these pre commit stages doesn’t pass.

From my experience using this in my own repo (https://github.com/nbQA-dev/nbQA) that's what happens

Reply:
Should we open an issue to remember to deduplicate these checks later?

Reply:
sure

Reply:
https://github.com/pandas-dev/pandas/issues/36642